codeowners_diff.py

```diff
@@ -88,10 +89,11 @@ def render_lines(self) -> Iterator[str]:
 
 def find_affected_paths(a: str, b: str) -> frozenset[str]:
     diff_generator = difflib.Differ().compare(a.splitlines(), b.splitlines())
+    pattern = re.compile(r'^[+-] *(?P<path>[^# ]+)')
```

Comment:
I opted to use a regex here to match changed lines only (`[+-]`), skip any spaces just in case (` *`), and then match anything that isn't a comment character or space (`[^# ]+`) to get the path. As we don't need the owners themselves, I ignored matching the remainder of the line.